src/components/Input/Input.tsx

```diff
@@ -1,129 +1,92 @@
 import * as React from "react";
 import bem from "../../utils/bem";
-import TextField, { TextFieldOptions } from "../TextField/TextField";
-import Label, { LabelOptions } from "../Label/Label";
-import HelperErrorText, {
-    HelperErrorTextOptions,
-} from "../HelperErrorText/HelperErrorText";
-
-export interface InputOptions {
-    baseClass?: string;
-    blockName?: string;
+import { InputTypes } from "./InputTypes";
 
-    labelId?: string;
-    labelContent?: JSX.Element;
+export interface InputProps {
+    /** ID that other components can cross reference for accessibility purposes */
+    id?: string;
 
-    inputId?: string;
-    inputType?: string;
-    inputValue?: string | number;
-    onInputChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    /** Populates the aria-label on the select */
+    ariaLabel?: string;
 
-    isRequired?: boolean;
+    /** Populates the aria-labelledby on the select */
+    ariaLabelledBy?: string;
 
-    helperContentOpts?: HelperErrorTextOptions;
-    errorContentOpts?: HelperErrorTextOptions;
+    /** Will add 'aria-required: true' to input */
+    required?: boolean;
 
-    showError?: boolean;
-}
-interface InputProps {
-    /** Optional baseClass for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
-    baseClass?: string;
-    /** Optional blockName for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
+    /** className you can add in addition to 'input' */
+    className?: string;
+
```

Comment:
does it make sense to put a catch-all attributes here like https://github.com/NYPL/nypl-design-system/blob/development/src/components/Link/Link.tsx#L26 since `input` is going to be flexible and general as well?

Reply:
edit:  huuh link attributes doesn't actually get used.  The idea I think is that if you want to attach something to the base html tag, like `<a>` or `<input>` it will give you a way to do that.

Reply:
I think it makes sense to add that in addition to className—the className you want added to the bem function specifically.

Reply:
I assume the link attributes are just getting passed right on through via the `linkProps`?